### PR TITLE
CMake fixes for new upolys

### PR DESCRIPTION
**Pull request by irislq**
When using SymEngine in an external C++ program the include files were not installed properly to include the new polys subfolder that was recently added. I changed CMakeList.txt to reflect there changes.

### CODE REVIEW

File: symengine/CMakeLists.txt [hunk 1]
```diff
@@ -99,9 +99,22 @@ install(TARGETS symengine
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
         )
-install(FILES ${HEADERS} "${symengine_BINARY_DIR}/symengine/symengine_config.h"
+
+# make install
+macro(INSTALL_HEADERS_WITH_DIRECTORY HEADER_LIST)
```
Comment (isuruf) [inline symengine/CMakeLists.txt]: No need to declare this as a macro, we can just use the body of the function

File: symengine/CMakeLists.txt [hunk 2]
```diff
@@ -69,16 +69,16 @@ endif()
 
 # Needed for "make install"
 set(HEADERS
-    add.h        symengine_assert.h  functions.h  monomials.h  polys/uintpoly.h polys/uexprpoly.h 
+    add.h        symengine_assert.h  functions.h  monomials.h polys/uintpoly.h polys/uexprpoly.h
```
Comment (isuruf) [inline symengine/CMakeLists.txt]: Can you change this back to the way it was?

File: symengine/CMakeLists.txt [hunk 3]
```diff
@@ -69,16 +69,16 @@ endif()
...
     basic.h      symengine_rcp.h     integer.h    mul.h        pow.h       constants.h
     symbol.h     expression.h        parser.h     mp_class.h   mp_wrapper.h
     basic-inl.h  dict.h           matrix.h     ntheory.h    rational.h complex.h
     visitor.h    eval_double.h    diophantine.h cwrapper.h  printer.h  real_double.h
     eval_mpfr.h  eval_arb.h       eval_mpc.h     complex_double.h         series_visitor.h
     real_mpfr.h  complex_mpc.h    type_codes.inc lambda_double.h series.h series_piranha.h
-    basic-methods.inc   series_flint.h  series_generic.h sets.h  derivative.h  polys/upolybase.h
-)
+    basic-methods.inc   series_flint.h  series_generic.h sets.h  derivative.h polys/upolybase.h
+)
```
Comment (isuruf) [inline symengine/CMakeLists.txt]: Same here

**Comment by isuruf** [comment]
Thanks for doing this. I've left a few comment. (minor changes)

**Comment by isuruf** [comment]
+1 to merge if tests pass

**Comment by srajangarg** [comment]
Thanks for this. I'm just wondering why the older version won't work?
`install(FILES ${HEADERS} DESTINATION include/symengine)`
It doesn't automatically create `polys/` when we tell it to place `polys/*.h` at `include/symengine`?

**Comment by irislq** [comment]
@srajangarg What that install does is create all the files in the `include/symengine` folder without preserving the directory structure, so everything goes into one place and it doesn't automatically create the `polys` folder. And then when you compile outside of SymEngine it can't find the header files from `polys` correctly.

**Comment by irislq** [comment]
@isuruf

**Pull request merged**